### DISCUSSION TITLE
Unable to receive GPS information on my Mac.

### DISCUSSION THREAD

**Discussion by KuranosukeOhta** [semuconsulting/PyGPSClient#117] [Q&A] [answered]
Hello. I am a high school student in Japan and a beginner in electronics, and I would like to display information on a USB GPS device with u-blox M8 (product link below) using PyGPSClient.
I have a USB connection to my Mac and the software seems to recognize the device as shown in the image, but I am unable to get correct location data.
I am wondering if there is some problem in the software environment, can you give me some advice?
Thank you in advance.

Product Link: https://www.tme.eu/mt/en/details/gu-902mgg-usb/gnss-gps-modules/yic/

![image](https://github.com/semuconsulting/PyGPSClient/assets/50015337/4d59e277-cd42-461b-a1fc-047852562d98)

**Comment by KuranosukeOhta** [comment, marked as answer]
I tried everything, but in the end it worked when I went out of the apartment!
Apparently the inside of the room was surrounded by steel bars, so it wasn't receiving a good signal.
I would like to use this software for various electronic work ideas (drones, rovers, etc.).
Thanks!

![image](https://github.com/semuconsulting/PyGPSClient/assets/50015337/77931b47-e7b2-4c21-9c89-d1056c3ece06)

**Comment by semuadmin** [comment]
Hi @LEGOnosuke 

Small integrated USB GNSS modules like yours have somewhat limited internal patch antennae which may not give great reception in urban areas. You may find using a small ferrous ground plane like this (or even just a steel saucepan lid) improves reception significantly:

https://www.sparkfun.com/products/17519

If your high school student budget allows 😉, an external antenna like this will offer much better results, but you’ll need a GNSS module with an SMA antenna connection. 

https://www.sparkfun.com/products/15192

e.g.

https://www.sparkfun.com/products/17285

Have fun!

**Reply by KuranosukeOhta** [reply]
Thanks for the detailed advice!
I see, the larger module is very practical and interesting. I think it would definitely be fun to combine this with ESP-32 and LoRa and other technologies to build a drone or rover.
Actually, I'm going to college this spring, so I'm looking forward to having more freedom to do research!
